Userland/Utilities/errno.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <stdlib.h>
+
+int main(int argc, char** argv)
+{
+    bool list = false;
+    bool search = false;
+    const char* keyword = nullptr;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(keyword, "Error number or string to search", "keyword", Core::ArgsParser::Required::No);
+    args_parser.add_option(list, "List all errno values", "list", 'l');
+    args_parser.add_option(search, "Search for error descriptions containing keyword", "search", 's');
+    args_parser.parse(argc, argv);
+
+    if (list) {
+        for (int i = 0; i < sys_nerr; i++) {
+            outln("{} {}", i, strerror(i));
+        }
+        return 0;
+    }
+
+    if (keyword == nullptr)
+        return 0;
+
+    if (search) {
+        for (int i = 0; i < sys_nerr; i++) {
+            auto error = String::formatted("{}", strerror(i));
+            if (error.contains(keyword, CaseSensitivity::CaseInsensitive)) {
+                outln("{} {}", i, error);
+            }
+        }
+        return 0;
+    }
+
+    auto error = String::formatted("{}", strerror(atoi(keyword)));
```

Comment:
As `atoi` is a terrible API and returns 0 for invalid input, this will lead to strange results for non-numeric values:

```
$ errno foo
foo Success (not an error)
```

Let's use `to_int()` instead, something like this:

```suggestion
    auto maybe_errno = StringView { keyword }.to_int();
    if (!maybe_errno.has_value()) {
        // print error?
        return 1;
    }
    auto error = String::formatted("{}", strerror(maybe_errno.value()));
```

Reply:
> `atoi` is a terrible API

yes

> this will lead to strange results for non-numeric values

yes, a bug

> Let's use `to_int()` instead

Done

![errno](https://user-images.githubusercontent.com/434827/118379309-cbd4ea00-b61c-11eb-85da-40d59be9bd57.png)